.github/workflows/directx-unit-test.yml

```diff
@@ -0,0 +1,65 @@
+name: DirectX unit test
+
+on:
+  push:
+    branches:
+    - main
+    - release/*
+  pull_request:
+    branches:
+    - main
+    - release/*
+
+jobs:
+  directx-unit-test:
+    name: DirectX unit test
+    runs-on: [self-hosted, windows, x64, win2004]
+    steps:
+    - name: Set up Python 3.8
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.8
+    - name: Set up Python 3 alias
+      run: |
+        $pythonDir = Split-Path (Get-Command python).Source
+        cd $pythonDir
+        echo @python %* > python3.bat
+      shell: pwsh
+    - name: Install make and git
+      shell: pwsh
+      run: |
+        if (Test-Path C:/ProgramData/chocolatey) {Remove-Item -Path C:/ProgramData/chocolatey -Recurse -Force}
+        Set-ExecutionPolicy Bypass -Scope Process -Force; [System.Net.ServicePointManager]::SecurityProtocol = [System.Net.ServicePointManager]::SecurityProtocol -bor 3072; iex ((New-Object System.Net.WebClient).DownloadString('https://chocolatey.org/install.ps1'))
+        choco install make git -y
+        echo "PATH=$env:PATH" | Out-File -FilePath $env:GITHUB_ENV -Append -Encoding utf8
+    - name: Install VS 2022 build tools
+      shell: pwsh
+      run: |
+        $env:SB_HOME=$PWD; ./dockerfile/directx/install-components.bat
+    - name: Checkout
+      uses: actions/checkout@v3
+    - name: Install dependencies
+      run: |
+        python -m pip install --no-build-isolation --user --upgrade pip wheel setuptools==65.7
+        python -m pip install --no-build-isolation --user .[test,nvworker]
+      shell: pwsh
+    - name: Build benchmarks
+      run: |
+        $env:SB_MICRO_PATH=$PWD
+        make directxbuild
+      shell: pwsh
+    - name: Build Third Party
+      run: |
+        $env:SB_MICRO_PATH=$PWD
+        make -C third_party directx_amd
```

Comment:
do you need to build in every run?

Reply:
if I do not build it then what if there's no bin for unit test?  I think we need to build here and control not build duplicate in makefile

Reply:
can you run the unit test in container? maybe just add a "docker run superbench/main:win2004 python setup.py test" task after docker build?